egg/core/trainers.py

```diff
@@ -173,9 +172,14 @@ def eval(self):
                     )
                 interaction = interaction.to("cpu")
                 mean_loss += optimized_loss
-                n_batches += 1
+
+                for callback in self.callbacks:
+                    callback.on_batch_end(
+                        interaction, optimized_loss, batch_id, is_training=False
+                    )
+
                 interactions.append(interaction)
-        mean_loss /= n_batches
+        mean_loss /= batch_id
```

Comment:
If you only have one batch wouldn't this raise a ZeroDivisionError?

Reply:
Would something like the following be ok?
```
 mean_loss /= (batch_id +1) 
```

Reply:
yep, that'd work. 
Any specific reason to remove n_batches though? I find it slightly more readable (it's personal/subjective I guess)

Reply:
Actually no, it just looked neater like this to me. I guess I will revert this change in the next push

Reply:
The original `n_batches` is set to zero at the start actually, so using this new formulation yields the same results.
Imo it is cleaner this way since:
- it mimics the behavior of the train method
- it reduces the number of line

Reply:
I don't think it yields the same formulation. `n_batches` is set to 0 but it is increased even in the first iteration https://github.com/robertodessi/EGG/blob/master/egg/core/trainers.py#L176
This is the same behavior for train and eval. I would just leave it as is with n_batches. Thanks!

Reply:
reverted in 2d25b57